funderfinder/utils/list_numfocus.py

```diff
@@ -0,0 +1,186 @@
+    )
+    soup = bs4.BeautifulSoup(page.text, features="html.parser")
+    for project_box in soup.find_all("div", class_="search-result-item-inner"):
+        link_parent = project_box.find("a", href=True)
+        link = link_parent["href"]
+        name = link_parent.parent.parent.text.strip()
+        project_page = requests.get(
+            link, headers=HEADERS, timeout=REQUESTS_TIMEOUT
+        ).text
+        github_ref = get_github_link(name, project_page)
+        if not github_ref:
+            try:
+                project_page_soup = bs4.BeautifulSoup(
+                    project_page, features="html.parser"
+                )
+                project_homepage_candidates = project_page_soup.find_all(
+                    string="Website"
+                )
+                project_homepage_elt = (
+                    None
+                    if not project_homepage_candidates
+                    else project_homepage_candidates[0]
+                )
+                if project_homepage_elt:
+                    project_homepage = project_homepage_elt.parent["href"]
+                    project_homepage_response = requests.get(
+                        project_homepage, headers=HEADERS, timeout=REQUESTS_TIMEOUT
+                    )
+                    github_ref = get_github_link(name, project_homepage_response.text)
+            except Exception as e:
+                LOGGER.warning(f"Exception when retrieving {link} for {name}: {e}")
+        projects.append(
+            {
+                "name": name,
+                "slug": link.strip().strip("/").split("/")[-1],
+                "github_name": github_ref,
+                "relationship": "sponsored",
+            }
+        )
+        # don't make requests any faster than every 2 seconds
+        time.sleep(2)
+    return projects
+
+
+def get_affiliated_projects() -> list:
+    """
+    Retrieve all numfocus affiliated projects
+    :return: List of project metadata as specified in module-level documentation
+    """
+    LOGGER.info("Retrieving affiliated projects")
+    projects = []
+    page = requests.get(
+        "https://numfocus.org/sponsored-projects/affiliated-projects",
+        headers=HEADERS,
+        timeout=REQUESTS_TIMEOUT,
+    )
+    soup = bs4.BeautifulSoup(page.text, features="html.parser")
+    for project_box in soup.find_all("div", class_="et_pb_blurb_content"):
+        link_parent = project_box.find("a", href=True)
+        link = link_parent["href"]
+        name = (
+            project_box.find("div", class_="et_pb_blurb_container")
+            .find("a")
+            .text.strip()
+        )
+        github_ref = get_github_link(name, link)
+        if not github_ref:
+            try:
+                project_page = requests.get(
+                    link, headers=HEADERS, timeout=REQUESTS_TIMEOUT
+                )
+                github_ref = get_github_link(name, project_page.text)
+            except Exception as e:
```

Comment:
nit: Like all python programmers, I also use bare exceptions. But the point of code review is to pretend like we're all perfect! (Right?) Do you expect a particular type of exception? If so, consider using that exception specifically.

Also, is there a reason you put two functions inside the try clause?

Reply:
Yeah that was laziness on my part, good point -- I've tightened to `requests.exceptions.RequestException`. There are a few different exceptions requests [might raise](https://requests.readthedocs.io/en/latest/user/quickstart/#errors-and-exceptions) but I don't think any of them require special treatment here.

As to the two functions, just because I thought it was neater to not have a separate if block after the exception block, but I'm willing to do that if you prefer!